docs/system-properties.adoc

```diff
@@ -44,24 +45,50 @@ Method-level configurations are visible in both `@BeforeEach` setup methods and
 Since v1.7.0, a class-level configuration means that the specified system properties are cleared/set before and reset after each individual test in the annotated class.
 ====
 
-Sometimes, you might also need to set a system property to a value that is not a constant expression, which is required for annotation values.
-In this case, you can still leverage the restore mechanism:
+== `@RestoreSystemProperties`
+`@RestoreSystemProperties` can be used to restore changes to system properties made directly in code.
+While `@ClearSystemProperty` and `@SetSystemProperty` set or clear specific properties and values, they don't allow property values to be calculated or parameterized, thus there are times you may want to directly set properties in your test code.
+`@RestoreSystemProperties` can be placed on test methods or test classes and will completely restore all system properties to their original state after a test or test class is complete.
+
+In this example, `@RestoreSystemProperties` is used on a test method, ensuring any changes made in that method are restored:
+
+[source,java,indent=0]
+----
+include::{demo}[tag=systemproperty_restore_test]
+----
+
+When `@RestoreSystemProperties` is used on a test class, any system properties changes made during the entire lifecycle of a test class, including test methods, `@BeforeAll`, `@BeforeEach` and 'after' methods, is restored after the test class lifecycle is complete.
+In addition, the annotation is inherited by each test method just as if each one was annotated with `@RestoreSystemProperties`.
+
+In the following example, `SomeOtherTest` is completely isolated from changes made in `MySystemPropertyRestoreTest` and `isolatedTest2` is isolated from changes made in `isolatedTest1`.
+Both `isolatedTest1` and `isolatedTest2` _will_ see system property values set in `@BeforeAll` and `@BeforeEach`.
 
 [source,java,indent=0]
 ----
-include::{demo}[tag=systemproperty_parameter]
+include::{demo}[tag=systemproperty_restore_class_level]
 ----
 
+Note that using `@RestoreSystemProperties` is not necessary to restore system properties modified via `@ClearSystemProperty` or `@SetSystemProperty`.
+However, it is OK and normal to use all three annotations together, which you might do if some property values are parameterized (i.e. need to be set in code) and others are not.
```

Comment:
Late addition to the docs:  The potential use of the three annotations together.

Reply:
An example for using the three annotation together has been added.